subsys/nfc/lib/Kconfig

```diff
@@ -21,6 +21,97 @@ config NFCT_IRQ_PRIORITY
 	  Sets NFC interrupt priority.
 	  Levels are from 0 (highest priority) to 6 (lowest priority)
 
+config NFC_THREAD_CALLBACK
+	bool "Thread context for NFC callbacks"
+	select RING_BUFFER
+	default y
+	help
+	  Use this option to decouple the user NFC callback
+	  from the NFC interrupt context. When enabled, the user
+	  callback is called from a thread (System Workqueue
+	  or a dedicated thread depending on the CONFIG_NFC_OWN_THREAD
+	  setting) instead of being called from the interrupt context.
+	  Using this option ensures robust NFC communication
+	  (deterministic time of the interrupt handling) and allows
+	  deferred callback execution, while it results in higher memory
+	  footprint and CPU load.
+
+if NFC_THREAD_CALLBACK
+
+config HEAP_MEM_POOL_SIZE
+	default 512
+
+config NFC_RING_SIZE
+	int "Size of ring buffer in bytes."
+	default 2048
+	help
+	  Size of the ring buffer.
+	  The specified size is expressed as n bytes.
+
+config NFC_LIB_CTX_MAX_SIZE
+	int "Maximum size of NFC library context in bytes."
+	default 16
+
+config NFC_OWN_THREAD
+	bool "Call NFC callback from its own thread"
+	default n
+
+if NFC_OWN_THREAD
+
+config NFC_THREAD_STACK_SIZE
+	int "Size of NFC thread stack"
+	default 256
+
+config NFC_THREAD_PRIORITY
+	int "Priority of NFC callback thread"
+	default -1
+
+endif# NFC_OWN_THREAD
+
+config NFC_ZERO_LATENCY_IRQ
+	bool "Use Zero Latency IRQ for NFC"
+	default y
+	select ZERO_LATENCY_IRQS
+	help
+	  Using this option ensures fast NFC interrupt
+	  execution on the cost of using additional interrupt (SWI).
+
+if NFC_ZERO_LATENCY_IRQ
+
+choice NFC_SWI
+	default NFC_SWI1
+	prompt "Software interrupt to use for NFC callback request."
+
+config NFC_SWI0
+	bool "SWI0"
+	depends on HAS_HW_NRF_EGU0
```

Comment:
I think `HAS_HW_NRF_SWI0` would be better as dependency. Please check it also for other configs in this choice

Reply:
But for nRF53 it should be `HAS_HW_NRF_EGU0`